Xamarin.Forms.Core/Span.cs

```diff
@@ -1,145 +1,101 @@
-
-			public void OnFontChanged(Font oldValue, Font newValue) =>
-				((IFontElement)_span).OnFontChanged(oldValue, newValue);
-
-			public void OnFontFamilyChanged(string oldValue, string newValue) =>
-				((IFontElement)_span).OnFontFamilyChanged(oldValue, newValue);
-
-			public void OnFontSizeChanged(double oldValue, double newValue) =>
-				((IFontElement)_span).OnFontSizeChanged(oldValue, newValue);
-
-			protected override void OnPropertyChanged(string propertyName = null)
-			{
-				base.OnPropertyChanged(propertyName);
-				_span.OnPropertyChanged(propertyName);
-			}
+			_mergedStyle = new MergedStyle(GetType(), this);
 		}
 
-		Color _backgroundColor;
-
-		BindableObject _fontElement;
-
-		Color _foregroundColor;
+		public static readonly BindableProperty StyleProperty = BindableProperty.Create("Style", typeof(Style), typeof(Span), default(Style),
+			propertyChanged: (bindable, oldvalue, newvalue) => ((Span)bindable)._mergedStyle.Style = (Style)newvalue);
 
-		string _text;
-
-		public Span()
+		public Style Style
 		{
-			_fontElement = new BindableSpan(this);
+			get { return (Style)GetValue(StyleProperty); }
+			set { SetValue(StyleProperty, value); }
 		}
 
+		public static readonly BindableProperty BackgroundColorProperty
+			= BindableProperty.Create(nameof(BackgroundColor), typeof(Color), typeof(Span), default(Color));
+
 		public Color BackgroundColor
 		{
-			get { return _backgroundColor; }
-			set
-			{
-				if (_backgroundColor == value)
-					return;
-				_backgroundColor = value;
-				OnPropertyChanged();
-			}
+			get { return (Color)GetValue(BackgroundColorProperty); }
+			set { SetValue(BackgroundColorProperty, value); }
 		}
 
-		[Obsolete("Font is obsolete as of version 1.3.0. Please use the Font properties directly.")]
-		public Font Font {
-			get { return (Font)_fontElement.GetValue(FontElement.FontProperty); }
-			set { _fontElement.SetValue(FontElement.FontProperty, value); }
+		[Obsolete("Foreground is obsolete as of version 2.6.0. Please use the TextColor property instead.")]
+		public static readonly BindableProperty ForegroundColorProperty
+			= BindableProperty.Create(nameof(ForegroundColor), typeof(Color), typeof(Span), default(Color), propertyChanged: ForegroundColorOnPropertyChanged);
+
+		static void ForegroundColorOnPropertyChanged(object bindable, object oldvalue, object newvalue)
+		{
+			((Span)bindable).TextColor = newvalue as Color? ?? default(Color);
 		}
 
 		public Color ForegroundColor
 		{
-			get { return _foregroundColor; }
-			set
-			{
-				if (_foregroundColor == value)
-					return;
-				_foregroundColor = value;
-				OnPropertyChanged();
-			}
+			get { return (Color)GetValue(ForegroundColorProperty); }
+			set { SetValue(ForegroundColorProperty, value); }
+		}
+
+		public static readonly BindableProperty TextColorProperty
+			= BindableProperty.Create(nameof(TextColor), typeof(Color), typeof(Span), default(Color));
+
+		public Color TextColor
```

Comment:
ForegroundColor was actually renamed to TextColor, which is why it has changed all over the UnitTests. ForegroundColor was then just added, but only as a relay to update TextColor.